tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/service/mapper/ProgrammeMembershipMapper.java

```diff
@@ -34,166 +32,187 @@ public class ProgrammeMembershipMapper {
-      ProgrammeMembership programmeMembership) {
-    ProgrammeMembershipDTO result = new ProgrammeMembershipDTO();
+      ProgrammeMembership entity) {
+    ProgrammeMembershipDTO dto = new ProgrammeMembershipDTO();
 
-    if (!programmeMembership.getCurriculumMemberships().isEmpty()) {
-      //Preserve backward-compatibility
-      result.setId(programmeMembership.getCurriculumMemberships().iterator().next().getId());
+    if (!entity.getCurriculumMemberships().isEmpty()) {
+      //Preserve backward-compatibility, set ID in DTO to the ID of the first curriculumMembership
+      dto.setId(entity.getCurriculumMemberships().iterator().next().getId());
     }
-    result.setUuid(programmeMembership.getUuid());
-    result.setProgrammeMembershipType(programmeMembership.getProgrammeMembershipType());
-    result.setProgrammeStartDate(programmeMembership.getProgrammeStartDate());
-    result.setProgrammeEndDate(programmeMembership.getProgrammeEndDate());
-    result.setLeavingReason(programmeMembership.getLeavingReason());
-    result.setAmendedDate(programmeMembership.getAmendedDate());
-    Programme programme = programmeMembership.getProgramme();
+
+    dto.setUuid(entity.getUuid());
+    dto.setProgrammeMembershipType(entity.getProgrammeMembershipType());
+    dto.setProgrammeStartDate(entity.getProgrammeStartDate());
+    dto.setProgrammeEndDate(entity.getProgrammeEndDate());
+    dto.setLeavingReason(entity.getLeavingReason());
+    dto.setAmendedDate(entity.getAmendedDate());
+    dto.setTrainingPathway(entity.getTrainingPathway());
+
+    Programme programme = entity.getProgramme();
     if (programme != null) {
-      result.setProgrammeId(programme.getId());
-      result.setProgrammeOwner(programme.getOwner());
-      result.setProgrammeName(programme.getProgrammeName());
-      result.setProgrammeNumber(programme.getProgrammeNumber());
-      result.setRotation(rotationToRotationDTO(programmeMembership.getRotation(), programme));
+      dto.setProgrammeId(programme.getId());
+      dto.setProgrammeOwner(programme.getOwner());
+      dto.setProgrammeName(programme.getProgrammeName());
+      dto.setProgrammeNumber(programme.getProgrammeNumber());
+      dto.setRotation(rotationToRotationDTO(entity.getRotation(), programme));
     }
-    result.setTrainingNumber(
-        trainingNumberToTrainingNumberDTO(programmeMembership.getTrainingNumber()));
 
-    if (programmeMembership.getPerson() == null) {
-      result.setPerson(null);
-    } else {
-      result.setPerson(personToPersonDTO(programmeMembership.getPerson()));
-    }
-    result.setTrainingPathway(programmeMembership.getTrainingPathway());
-    if (result.getUuid() != null) {
+    dto.setTrainingNumber(
+        trainingNumberMapper.trainingNumberToTrainingNumberDTO(entity.getTrainingNumber()));
+    dto.setPerson(personToPersonDTO(entity.getPerson()));
+
+    dto.setCurriculumMemberships(Lists.newArrayList());
```

Comment:
I noticed that the [docs](https://guava.dev/releases/31.1-jre/api/docs/com/google/common/collect/Lists.html#newArrayList()) say to use the ArrayList constructor in Java 7+